src/twisted/internet/tcp.py

```diff
@@ -669,7 +672,19 @@ def _resolveIPv6(ip, port):
     @raise socket.gaierror: if either the IP or port is not numeric as it
         should be.
     """
-    return socket.getaddrinfo(ip, port, 0, 0, 0, _NUMERIC_ONLY)[0][4]
+    usedPort = port
+    if isinstance(port, int):
+        # On macOS `getaddrinfo` will reject invalid port number while other
```

Comment:
Thanks @glyph for the extra info.

Now it makes sense.

---------

I will write a separate check to make sure this will continue to fail on macOS

This code was written a long time ago with support for python 2.6

I now see that Linux is also failing

macos

```
% python3 
Python 3.9.6 (default, Nov 10 2023, 13:38:27) 
[Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> _AI_NUMERICSERV = getattr(socket, "AI_NUMERICSERV", 0)
>>> socket.getaddrinfo('127.0.0.1', -1, 0, 0, 0, _AI_NUMERICSERV)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/socket.py", line 953, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
>>> 

```


linux
```
$ python3
Python 3.11.9 (main, Apr  6 2024, 17:59:24) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> _AI_NUMERICSERV = getattr(socket, "AI_NUMERICSERV", 0)
>>> socket.getaddrinfo('127.0.0.1', -1, 0, 0, 0, _AI_NUMERICSERV)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -8] Servname not supported for ai_socktype
>>> 
```


--------

On Windows, no exception is raised

```
> python.exe
Python 3.11.7 (tags/v3.11.7:fa7a6f2, Dec  4 2023, 19:24:49) [MSC v.1937 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> _AI_NUMERICSERV = getattr(socket, "AI_NUMERICSERV", 0)
>>> socket.getaddrinfo('127.0.0.1', -1, 0, 0, 0, _AI_NUMERICSERV)
[(<AddressFamily.AF_INET: 2>, 0, 0, '', ('127.0.0.1', 65535))]
```

Reply:
> I now see that Linux is also failing

this kind of detail is exactly why i want to have a mechanical record of the behavior, rather than just a comment :-). Thanks for checking!

Reply:
Done. Let me know if this is what you were expecting.

I am not used to write this type of tests... as this is a test for a stdlib function, not our code.

We have tests for _resolveIPv6 with invalid port number, and for me that is all that we should care about :)